starfish/image/_segmentation/watershed.py

```diff
@@ -64,76 +103,217 @@ def show(self, figsize: Tuple[int, int]=(10, 10)) -> None:
+        """Implements watershed segmentation of cells seeded from a DAPI image the the point clouds
+
+        Watershed segmentation proceeds by constructing basins that extend from the DAPI image
+        and the point cloud constructed from a maximum projection across rounds and channels. It
+        constructs a mask that extends from the basins to prevent non-cellular regions from being
+        included.
+
+        Parameters
+        ----------
+        dapi_img : np.ndarray[np.float32]
+            nuclei image
+        stain_img : np.ndarray[np.float32]
+            stain image
+        """
         self.dapi = dapi_img / dapi_img.max()
         self.stain = stain_img / stain_img.max()
 
-        self.dapi_thresholded = None
+        self.dapi_thresholded: Optional[np.ndarray] = None  # dtype: bool
         self.markers = None
-        self.num_cells = None
+        self.num_cells: Optional[int] = None
         self.mask = None
         self.segmented = None
 
     def segment(
-            self, dapi_thresh, stain_thresh, size_lim, disk_size_markers=None, disk_size_mask=None,
-            min_dist=None
-    ):
+            self,
+            dapi_thresh: Number,
+            stain_thresh: Number,
+            size_lim: Tuple[int, int],
+            disk_size_markers: Optional[int]=None,  # TODO ambrosejcarr what is this doing?
+            disk_size_mask: Optional[int]=None,  # TODO ambrosejcarr what is this doing?
+            min_dist: Optional[Number]=None
+    ) -> np.ndarray:
+        """Execute watershed cell segmentation.
+
+        Parameters
+        ----------
+        dapi_thresh, stain_thresh : Number
+            Threshold for the dapi and stain images. All pixels with intensities above this size
+            will be considered part of the objects in the respective images
+        size_lim : Tuple[int, int]
+            min and max allowable size for nuclei objects in the dapi_image
+        disk_size_markers : Optional[int]
+            if provided ...
+        disk_size_mask : Optional[int]
+            if provided ...
+        min_dist : Optional[int]
+            if provided, nuclei within this distance of each other are combined into single
+            objects.
+
+        Returns
+        -------
+        np.ndarray[int32] :
+            label image with same size and shape as self.dapi_img
+        """
         min_allowed_size, max_allowed_size = size_lim
         self.dapi_thresholded = self.filter_dapi(dapi_thresh, disk_size_markers)
         self.markers, self.num_cells = self.label_nuclei(
+            self.dapi_thresholded,
             min_allowed_size, max_allowed_size, min_dist
         )
         self.mask = self.watershed_mask(stain_thresh, self.markers, disk_size_mask)
         self.segmented = self.watershed(self.markers, self.mask)
         return self.segmented
 
-    def filter_dapi(self, dapi_thresh, disk_size):
+    def filter_dapi(self, dapi_thresh: float, disk_size: Optional[int]) -> np.ndarray:
```

Comment:
should be called filter_nuclei. in general, we should not refer to dapi, as there are other ways to stain nuclei, e.g., drac5 as employed by Dartfish.